### PR TITLE
Update settings_backup.php

### PR DESCRIPTION
**Pull request by aftechro**
Backup and restore from the backup menu. Add backups folder into the uploads, so the backup path will be uploads/backups

https://www.veed.io/view/5bc75a75-9af6-4fc4-a462-6a161c8b9c23?panel=share

### CODE REVIEW

**Comment by wrongecho** [comment]
Test these changes at: https://patch14852.pr-review.itflow.org  
 _(automatic message)_

**Comment by aftechro** [comment]
auch, not sure about the security there, but if one of you guys can have a look at it, please do so, thank you!

File: settings_backup.php [hunk 1]
```diff
@@ -1,37 +1,220 @@
...
+
+// Handle backup action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['backup'])) {
+    // Create a backup
+    $backupFileName = date("d-m-Y_H-i-s") . ".sql";
+    $backupPath = $backupFolder . $backupFileName;
+
+    // Run mysqldump command to include table content
+    $escapedBackupPath = escapeshellarg($backupPath);
+    $command = "mysqldump --complete-insert --skip-comments -h $dbhost -u $dbusername -p$dbpassword $database > $escapedBackupPath";
```
[the remark this answers is not in the document]
Comment (aftechro) [reply settings_backup.php:22]: false positive?

File: settings_backup.php [hunk 2]
```diff
@@ -1,37 +1,220 @@
...
+}
+
+
+
+// Handle delete selected action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['delete-selected'])) {
+    // Implement delete selected logic here
+    if (isset($_POST['selectedBackups'])) {
+        foreach ($_POST['selectedBackups'] as $selectedBackup) {
+            unlink($backupFolder . $selectedBackup);
```
[the remark this answers is not in the document]
Comment (aftechro) [reply settings_backup.php:73]: have no idea what this means

File: settings_backup.php [hunk 3]
```diff
@@ -1,37 +1,220 @@
...
+    // Refresh backup list after creating a new backup
+    $backups = array_diff(scandir($backupFolder), array('..', '.'));
+}
+
+// Handle restore action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['proceed-restore'])) {
+    $selectedBackup = $_POST['proceed-restore'];
+
+    $sqlFile = $backupFolder . $selectedBackup;
+    $sqlContent = file_get_contents($sqlFile);
```
[the remark this answers is not in the document]
Comment (aftechro) [reply settings_backup.php:34]: again, not sure what this means

**Comment by wrongecho** [comment]
Hey! It's been a while :)

I can't actually get this to load on PR Review?

**Comment by aftechro** [comment]
you`ll need to create backups folder inside uploads so path will be uploads/backups.  test it locally first :D

**Comment by wrongecho** [comment]
Ah okay. Is there a way we can do that within the page itself, possibly with `mkdirMissing`?

**Comment by aftechro** [comment]
not sure the best approach ( folder structure and security wise), that`s why i`ve asked for second opinion :D i`ve done the script locally, test it and works. now, going from this, sure, we can make it much better, at least we have a base to work from. still need to figure out the restore from file aspect, which should be another modal uploading file and restore script to handle that file. But with baby steps, and all the great minds, we`ll get there :D

**Pull request closed without merging**